lxd/container_lxc.go

```diff
@@ -3508,14 +3522,18 @@ func (c *containerLXC) getDiskLimits() (map[string]deviceBlockLimit, error) {
 		device := deviceBlockLimit{readBps: readBps, readIops: readIops, writeBps: writeBps, writeIops: writeIops}
 
 		for _, block := range blocks {
-			dev := strings.TrimPrefix(block, "/dev/")
+			if !strings.HasPrefix(block, "/dev/") {
+				continue
+			}
 
+			dev := strings.TrimPrefix(block, "/dev/")
 			if strings.Contains(dev, "/") {
 				continue
 			}
 
 			if !shared.PathExists(fmt.Sprintf("/sys/class/block/%s/dev", dev)) {
-				return nil, fmt.Errorf("Disk %s is missing /sys/class/block entry", dev)
+				shared.Log.Error("Disk is missing /sys/class/block entry", log.Ctx{"device": dev})
+				continue
```

Comment:
Does this mean that it's possible that a user asked for a quota on a block device and if we couldn't set it they'll only get an error in the log? I agree it shouldn't fail if they haven't asked for a quota (i.e. hasDiskLimits above seems reasonable), but I think we should fail to start a container if we can't apply disk limits and the user has asked for them.